hs_file_types/models/base.py

```diff
@@ -612,6 +722,158 @@ def get_allowed_storage_file_types(cls):
+                msg = msg.format(path_to_check)
+                raise ValidationError(msg)
+
+            # get the files from the specified folder location
+            res_files = ResourceFile.list_folder(resource=resource, folder=folder_path,
+                                                 sub_folders=False)
+            if not res_files:
+                msg = "The specified folder {} does not contain any file."
+                msg = msg.format(path_to_check)
+                raise ValidationError(msg)
+            else:
+                # check if the specified folder is suitable for aggregation
+                if cls.check_files_for_aggregation_type(res_files):
+                    # get the primary file suitable for creating a specific aggregation type
+                    res_file = cls.get_primary_resouce_file(res_files)
+                else:
+                    res_file = None
+
+        if res_file is None or not res_file.exists:
+            raise ValidationError("File not found.")
+
+        if res_file.has_logical_file:
+            msg = "Selected {} {} is already part of an aggregation."
+            if folder_path is None:
+                msg = msg.format('file', res_file.file_name)
+            else:
+                msg = msg.format('folder', folder_path)
+            raise ValidationError(msg)
+
+        return res_file, folder_path
+
+    @classmethod
+    def get_primary_resouce_file(cls, resource_files):
+        """Returns one specific file as the primary file from the list of resource
+        files *resource_files*. A file is a primary file which can be used for creating a
+        file type (aggregation). Subclasses must implement this.
+
+        :param  resource_files: a list of resource files - instances of ResourceFile
+        :return a resource file (instance of ResourceFile) if found, otherwise, None
+        """
+
+        raise NotImplementedError
+
+    @staticmethod
+    def get_aggregation_display_name():
+        """Sub classes must implement this method to return a name for this
+        logical (aggregation) type used in UI"""
+        raise NotImplementedError()
+
+    @staticmethod
+    def _cleanup_on_fail_to_create_aggregation(user, resource, folder_to_delete, original_folder,
+                                               aggregation_from_folder):
+        """Deletes folder if a new aggregation folder *folder_to_delete*  was created
+        :param  user: user hwo was trying to create the aggregation
+        :param  resource: an instance of CompositeResource for which the aggregation was created
+        :param  folder_to_delete: the aggregation folder to delete
+        :param  original_folder: folder path for the files originally located before moving to the
+        new aggregation folder
+        :param  aggregation_from_folder: (bool) a flag to indicate if the aggregation was being
+        created from a folder or not
+        """
+        # had to import it here to avoid import loop
```

Comment:
Can you explain where the import loop is?  I believe putting the import here is more of a bandaid than a solution.  If you can explain the loop here, then maybe later on we can work on refactoring.

Reply:
```
 File "/usr/local/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/hydroshare/hs_file_types/models/__init__.py", line 1, in <module>
    from base import AbstractLogicalFile, AbstractFileMetaData      # noqa
  File "/hydroshare/hs_file_types/models/base.py", line 31, in <module>
    from hs_core.views.utils import remove_folder, move_or_rename_file_or_folder
  File "/hydroshare/hs_core/views/__init__.py", line 46, in <module>
    from . import resource_rest_api
  File "/hydroshare/hs_core/views/resource_rest_api.py", line 27, in <module>
    from hs_core.views import serializers
  File "/hydroshare/hs_core/views/serializers.py", line 11, in <module>
    RESOURCE_TYPES = [rtype.__name__ for rtype in utils.get_resource_types()]
  File "/hydroshare/hs_core/hydroshare/utils.py", line 53, in get_resource_types
    for model in apps.get_models():
  File "/usr/local/lib/python2.7/site-packages/django/utils/lru_cache.py", line 101, in wrapper
    result = user_function(*args, **kwds)
  File "/usr/local/lib/python2.7/site-packages/django/apps/registry.py", line 168, in get_models
    self.check_models_ready()
  File "/usr/local/lib/python2.7/site-packages/django/apps/registry.py", line 131, in check_models_ready
    raise AppRegistryNotReady("Models aren't loaded yet.")
django.core.exceptions.AppRegistryNotReady: Models aren't loaded yet.
```